test/unit-test/test_from_erlang.ml

```diff
@@ -0,0 +1,106 @@
+open Base
+open Fialyzer
+open Ast_intf
+open From_erlang
+open Obeam.Abstract_format
+
+let%expect_test "from_erlang" =
+  let print abstract_format =
+    Variable.reset_count ();
+    expr_of_erlang_expr abstract_format
+    |> [%sexp_of: expr]
+    |> Expect_test_helpers_kernel.print_s in
+
+
+  (* X *)
+  print (ExprVar(1, "X"));
+  [%expect {|
+    (Var X)
+  |}];
+
+  (* {X, Y, Z} *)
+  print (ExprTuple(1, [ExprVar(1, "X"); ExprVar(1, "Y"); ExprVar(1, "Z")]));
+  [%expect {|
+    (Tuple (
+      (Var X)
+      (Var Y)
+      (Var Z)))
+  |}];
+
+
+  (*
+   * fun (X) -> X end
+   *)
+  print (ExprFun(1, None, [
+    ClsFun(1, [PatVar(1, "X")], None, ExprVar(1, "X")) 
+  ]));
+  [%expect {|
+    (Letrec
+      ((
+        __A__ (
+          Abs
+          (__B__)
+          (Case
+            (Tuple ((Var __B__)))
+            ((((PatTuple ((PatVar X))) (Constant (Atom true))) (Var X)))))))
+      (Var __A__))
+  |}];
+
+  (*
+   * fun F(X) -> F(X) end
+   *)
+  print (ExprFun(1, Some("F"), [
+    ClsFun(1, [PatVar(1, "X")], None, ExprLocalCall(1, ExprVar(1, "F"), [ExprVar(1, "X")])) 
+  ]));
+  [%expect {|
+    (Letrec
+      ((
+        F (
+          Abs
+          (__B__)
+          (Case
+            (Tuple ((Var __B__)))
+            ((
+              ((PatTuple ((PatVar X))) (Constant (Atom true)))
+              (App (Var F) ((Var X)))))))))
+      (Var F))
+  |}];
```

Comment:
This conversion looks to be too complicated.
Can this be converted as follows?
```ocaml
Abs (X) (Var X)
```
In order to realize this, case division may be necessary.
- [FunName is None]: convert just `Abs` expression
- [FunName is not exists]: convert `Letrec` expression
- [Single case and all arguments are PatVar or PatUniverse]: No pattern matching
- [Multi cases or other patterns]: `Case` expression

Reply:
OK.  To simplify the conversion itself, I did it.  However, you maybe right. So, I'll do so.

Reply:
👍